python/ribasim/ribasim/model.py

```diff
@@ -376,6 +448,68 @@ def reset_contextvar(self) -> "Model":
         context_file_loading.set({})
         return self
 
+    def max_node_id(self) -> int:
+        return self.network.node.df.index.max()
+
+    def max_allocation_network_id(self) -> int:
+        m = self.network.node.df.allocation_network_id.max()
+        if pd.isna(m):
+            m = 0
+        return m
+
+    def merge_model(
+        self,
+        model_added: "Model",
+        offset_node_id: int | None = None,
+        offset_allocation_network_id: int | None = None,
+        offset_spacial: tuple[float, float] = (0.0, 0.0),
+        inplace: bool = True,
+    ):
+        """
+        Merge copies of the nodes and edges of an added model into this model.
+        The added model is not modified, but the following modificadions are made to the added data:
```

Comment:
```suggestion
        The added model is not modified, but the following modifications are made to the added data:
```

A spellcheck in your IDE might help you 😄